### ISSUE TITLE
Additional properties not ignored on $refs against OpenAPI 3.0 spec

### ISSUE THREAD

**Issue by PetrusAsikainen** [spec-first/connexion#1721]
### Description

OpenAPI 3.0 [Reference Objects](https://swagger.io/specification/v3/#reference-object) specify that all properties outside `$ref` must be ignored. This is inherited from [JSON Schema Draft 4(/5)](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-00#section-7) which says _"All other properties in a "$ref" object MUST be ignored."_

The jsonschema package [uses `applicable_validators=ignore_ref_siblings`](https://github.com/python-jsonschema/jsonschema/blob/main/jsonschema/validators.py#L646) to follow the spec.

Connexion then adds the OpenAPI-specific changes to `jsonschema` [using their own `extend` function](https://github.com/spec-first/connexion/blob/main/connexion/json_schema.py#L138).

However, extend [drops `applicable_validators`](https://github.com/python-jsonschema/jsonschema/blob/main/jsonschema/validators.py#L493), which makes Connexion's validation consider any extra properties again!

I've filed a bug as python-jsonschema/jsonschema#1125, but it's affecting `connexion` as a consumer of `extend` as well.

### Steps to reproduce

```python
import jsonschema, connexion

jsonschema.validate(
  schema={
    "$defs": { "test": { "type": "number" } },
    "$ref": "#/$defs/test",
    "maximum": 1
  },
  instance=2,
  cls=connexion.json_schema.Draft4RequestValidator
)
```

### Expected behaviour

No error, since only `"type": "number"` is validated.

### Actual behaviour

```plaintext
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/petrus/acs2/venv/lib/python3.11/site-packages/jsonschema/validators.py", line 1121, in validate
    raise error
jsonschema.exceptions.ValidationError: 2 is greater than the maximum of 1

Failed validating 'maximum' in schema:
    {'$defs': {'test': {'type': 'number'}},
     '$ref': '#/$defs/test',
     'maximum': 1}

On instance:
    2
```

### Additional info:

- Python 3.11.2
- connexion 2.14.2

**Comment by PetrusAsikainen** [comment]
Hmm, probably now fixed since `jsonschema` fixed this quickly in 4.18.3 and `connexion` pulls in `>= 4.0.1`

**Comment by RobbeSneyders** [comment]
Thanks for raising this both here and at jsonschema @PetrusAsikainen.
